Source/multi.cpp

```diff
@@ -724,60 +714,59 @@ int NetInit(int bSinglePlayer, int *pfExitProgram)
 		UiData.profilebitmapcallback = (void(*)())UiProfileDraw;
 		UiData.profilecallback = (void(*)())UiProfileCallback;
 		UiData.profilefields = UiProfileGetString();
-		memset(sgbPlayerTurnBitTbl, 0, 4u);
-		gbGameDestroyed = 0;
-		memset(sgbPlayerLeftGameTbl, 0, 4u);
-		memset(sgdwPlayerLeftReasonTbl, 0, 0x10u);
-		memset(sgbSendDeltaTbl, 0, 4u);
-		memset(plr, 0, 0x15360u);
-		memset(sgwPackPlrOffsetTbl, 0, 8u);
+		memset(sgbPlayerTurnBitTbl, 0, sizeof(sgbPlayerTurnBitTbl));
+		gbGameDestroyed = FALSE;
+		memset(sgbPlayerLeftGameTbl, 0, sizeof(sgbPlayerLeftGameTbl));
+		memset(sgdwPlayerLeftReasonTbl, 0, sizeof(sgdwPlayerLeftReasonTbl));
+		memset(sgbSendDeltaTbl, 0, sizeof(sgbSendDeltaTbl));
+		memset(plr, 0, sizeof(plr));
+		memset(sgwPackPlrOffsetTbl, 0, sizeof(sgwPackPlrOffsetTbl));
 		SNetSetBasePlayer(0);
-		if (v14)
-			v4 = multi_init_single(&ProgramData, &a2, &UiData);
-		else
-			v4 = multi_init_multi(&ProgramData, &a2, &UiData, a4);
-		if (!v4)
-			return 0;
-		sgbNetInited = 1;
-		sgbTimeout = 0;
+		if (bSinglePlayer) {
+			if (!multi_init_single(&ProgramData, &plrdata, &UiData))
+				return FALSE;
+		} else {
+			if (!multi_init_multi(&ProgramData, &plrdata, &UiData, pfExitProgram))
+				return FALSE;
+		}
+		sgbNetInited = TRUE;
+		sgbTimeout = FALSE;
 		delta_init();
 		InitPlrMsg();
 		buffer_init(&sgHiPriBuf);
 		buffer_init(&sgLoPriBuf);
-		dword_678628 = 0;
+		dword_678628 = FALSE;
 		sync_init();
 		nthread_start(sgbPlayerTurnBitTbl[myplr]);
 		dthread_start();
-		MI_Dummy(0); /* v5 */
+		break_exception();

```

Comment:
Are we sure that this should be `break_exception`? Could it not be another function that has a NOP body?

Reply:
we have no idea what function this was, any suggestions how to handle it?

Reply:
Perhaps add a `dummy_nop` function that we can later try to correlate with other versions, to find it's actual implementation. You could even name it `dummy_nop_used_in_NetInit` or something.

Reply:
I'm ready to merge this once we add a `dummy_nop_used_in_NetInit` function :) Also, I don't expect it to be added soon :) @AJenbo enjoy your vacation! :)